sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/ReceiverTests.cs

```diff
@@ -121,6 +121,22 @@ public void ReceiveValidatesMaxWaitTime()
                 Throws.InstanceOf<ArgumentOutOfRangeException>());
         }
 
+        [Test]
+        public void ReceiveValidatesMaxWaitTimePrefetchMode()
+        {
+            var account = Encoding.Default.GetString(GetRandomBuffer(12));
+            var fullyQualifiedNamespace = new UriBuilder($"{account}.servicebus.windows.net/").Host;
+            var connString = $"Endpoint=sb://{fullyQualifiedNamespace};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey={Encoding.Default.GetString(GetRandomBuffer(64))}";
+            var client = new ServiceBusClient(connString);
+            var receiver = client.CreateReceiver("queue", options: new ServiceBusReceiverOptions { PrefetchCount = 10 });
+            Assert.That(
```

Comment:
Is this right? Shouldn't this not throw an exception with the changes in this PR?

Reply:
It will throw, but not ArgumentOutOfRangeException.
We aren't creating service buses in that test class, so exception will be: "Azure.Messaging.ServiceBus.ServiceBusException: No such host is known ErrorCode: HostNotFound (ServiceCommunicationProblem) ---> System.Net.Sockets.SocketException: No such host is known
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()"